bcipy/tasks/tests/test_main_frame.py

```diff
@@ -0,0 +1,100 @@
+        self.assertTrue(np.all(self.decision_maker.list_epoch[-1]['list_distribution'][-1] == probability_distribution))
+        #print(self.decision_maker.list_epoch[-1]['list_distribution'][-1])
+        #print(probability_distribution)
+        self.assertFalse(decision)
+
+    def test_decide_with_commit(self):
+        """Test decide method with case of commit"""
+        probability_distribution = np.ones(len(self.decision_maker.alphabet))
+        decision, chosen_stimuli = self.decision_maker.decide(probability_distribution)
+        #self.assertTrue(np.all(self.decision_maker.list_epoch[-1]['list_distribution'][-1] == probability_distribution))
+        #print(self.decision_maker.list_epoch[-1]['list_distribution'][-1])
+        #print(probability_distribution)
+        #print(self.decision_maker.sequence_counter)
+        #print(self.decision_maker.min_num_seq)
+        self.assertTrue(decision)
+        self.assertEqual(chosen_stimuli,[])
+
+    def test_update_with_letter(self):
+        """Test update method with letter being the new state"""
+        old_displayed_state = self.decision_maker.displayed_state
+        old_state = self.decision_maker.state
+        new_state = 'E'
+        self.decision_maker.update(state=new_state)
+        self.assertEqual(self.decision_maker.state,old_state+new_state)
+        self.assertEqual(self.decision_maker.displayed_state,old_displayed_state+'E')
+
+    def test_update_with_backspace(self):
+        """Test update method with backspace being the new state"""
+        old_displayed_state = self.decision_maker.displayed_state
+        old_state = self.decision_maker.state
+        new_state = '<'
+        self.decision_maker.update(state=new_state)
+        self.assertEqual(self.decision_maker.state,old_state+new_state)
+        """Check that displayed state does not take into account the backspace """
+        self.assertEqual(self.decision_maker.displayed_state,old_displayed_state[0:-1])
+        self.assertLess(len(self.decision_maker.displayed_state),len(self.decision_maker.state))
+
+    def test_reset(self):
+        """Test reset of decision maker state"""
+        self.decision_maker.reset()
+        self.assertEqual(self.decision_maker.state,'')
+        self.assertEqual(self.decision_maker.displayed_state,'')
+        self.assertEqual(self.decision_maker.time,0)
+        self.assertEqual(self.decision_maker.sequence_counter,0)
+
+    def test_form_display_state(self):
+        """Test form display state method with a dummy state"""
+        self.decision_maker.update(state='ABC<.E')
+        self.decision_maker.form_display_state(self.decision_maker.state)
+        self.assertEqual(self.decision_maker.displayed_state,'ABE')
+        self.decision_maker.reset()
+
+    def test_do_epoch(self):
+        """Test do_epoch method"""
+        self.decision_maker.do_epoch()
+        self.assertEqual(self.decision_maker.sequence_counter,0)
+        print(self.decision_maker.state)
+'''
```

Comment:
?

Reply:
Whoops, that was there for debugging purposes. Meant to delete. Just did!